tests/integration/pdo/mysql/base-class/test_prepared_stmt_1.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*DESCRIPTION
+When PDO base class constructor is used to create connection object
+and a query is executed via PDOStatement::execute(), the agent should
+ - not generate errors
+ - record a datastore metrics
+ - record a datastore span event
+Moreover, when the query execution time exceeds the explain threshold,
+the agent should record a slow sql trace with explain plan.
+*/
+
+/*SKIPIF
+<?php require(realpath (dirname ( __FILE__ )) . '/../../skipif_mysql.inc');
+*/
+
+/*ENVIRONMENT
+DATASTORE_PRODUCT=MySQL
+DATASTORE_COLLECTION=tables
+*/
+
+/*INI
+;comment=Set explain_threshold to 0 to ensure that the slow query is recorded.
+newrelic.transaction_tracer.explain_threshold = 0
+;comment=Disable instance and database name reporting (it is tested separately).
```

Comment:
What's the advantage of disabling instance and database name reporting here?
It probably makes sense to have at least one test with instance and database name reporting disabled to ensure everything else still works, but since it's on by default, wouldn't it make sense to leave it on to ensure interoperability even if we aren't explicitly testing or looking for metrics for instance and database name reporting?

Reply:
> What's the advantage of disabling instance and database name reporting here?

It makes maintaining these tests easier - these tests worry only about metrics, slow sql and span events, not so much about values of database instance attributes in the slow sql and span events. These tests also test functionality of disabling instance reporting - datastore instance attributes are not present in neither slow sql trace nor in span event.

> It probably makes sense to have at least one test with instance and database name reporting disabled to ensure everything else still works, but since it's on by default, wouldn't it make sense to leave it on to ensure interoperability even if we aren't explicitly testing or looking for metrics for instance and database name reporting?

There are tests dedicated to testing database instance reporting specifically (test_instance_reporting) and those tests test all possible and supported scenarios of the values of those attributes.

Testing too many things at once makes the tests harder to understand and harder to maintain. Slow sql functionality is different from datastore instance reporting functionality and that is why I decided to test them separately.